### PR TITLE
Fixed access code message bug and discount bug on pre-payment page

### LINKED ISSUE #2972: Access codes not working - giving error "Promotional Code not valid"
After the organizer has set up an access code and new attendees make an order with this access code, the error message comes up "The promotional code entered is not valid. No offer has been applied to this order."

Expected: User can make an order of the hidden ticket.

screenshot from 2017-01-18 22 26 48

### PR DESCRIPTION
**Pull request by magdalenesuo**
fixed #2972 

**Both Access And Discount Codes** Applied:
screenshot from 2017-01-19 15-46-06

Only **Access Code** Applied:
screenshot from 2017-01-19 16-18-25

Only **Discount Code** Applied:
screenshot from 2017-01-19 15-51-33

**Invalid Code**:
screenshot from 2017-01-19 15-47-53

### CODE REVIEW

File: app/helpers/ticketing.py [hunk 1]
```diff
@@ -240,16 +240,25 @@ def create_order(form, from_organizer=False):
...
+
+            if access_code and discount:
+                order.discount_code = discount
+                flash('Both access code and discount code has been applied. You can make this order now.', 'success')
+            elif discount:
                 order.discount_code = discount
-                flash('The promotional code entered is valid.offer has been applied to this order.', 'success')
+                flash('The promotional code entered is valid. Offer has been applied to this order.', 'success')
+            elif access_code:
+                flash('You access code is applied and you can make this order now.', 'success')
```
Comment (SaptakS) [inline app/helpers/ticketing.py]: It should be "Your" I think.

File: app/helpers/ticketing.py [hunk 2]
```diff
@@ -240,16 +240,25 @@ def create_order(form, from_organizer=False):
         discount=None
         if ticket_discount:
             discount=TicketingManager.get_discount_code(form.get('event_id'), form.get('promo_code',''))
-            if not discount:
-                flash('The promotional code entered is not valid. No offer has been applied to this order.', 'danger')
-            else:
+            access_code = TicketingManager.get_access_code(form.get('event_id'), form.get('promo_code', ''))
+
```
Comment (SaptakS) [inline app/helpers/ticketing.py]: Remove the newline please. Thanks.

**Comment by magdalenesuo** [comment]
@SaptakS made changes.

**Comment by mariobehling** [comment]
@SaptakS Please merge dev to master. So, we can have the fix on the live site. Thank you!

**Comment by SaptakS** [comment]
@mariobehling, I think @magdalenesuo is working on https://github.com/fossasia/open-event-orga-server/issues/2975. I will merge development once that is fixed. That is also essential. I am hoping she will complete by tonight.

**Pull request merged**